Addons/tsgExoticQuadrature.hpp

```diff
@@ -178,29 +193,62 @@ inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double sh
                                     std::move(weights_cache), description);
 }
 
-// Basic function overloads.
+/*
+ * \ingroup TasmanianAddonsExoticQuad
+ * \brief Constructs a CustomTabulated object containing the exotic quadrature nodes and quadrature weights for a given level \b n,
+ * scalar \b shift, a reference TasmanianSparseGrid \b grid containing loaded values of the weight function, a string \b description,
+ * and an optional bool \b is_symmetric that should be set to true if the weight function is symmetric.
+ *
+ * It is assumed that grid is a sparse grid interpolant/surrogate to the weight function. Only the nodes in grid.getLoadedPoints(),
+ * the quadrature weights in grid.getQuadratureWeights(), and the loaded weight function values in grid.getLoadedValues() are used
+ * in the computation of the exotic quadrature. Specifically, these vectors are used to orthogonalize the polynomial basis needed
+ * to compute the optimal exotic quadrature.
+ */
+inline TasGrid::CustomTabulated getExoticQuadrature(const int n, const double shift, const TasGrid::TasmanianSparseGrid &grid,
+                                                    const char* description, const bool is_symmetric = false) {
+    if (grid.getPoints().size() == 0) {
```

Comment:
Important points:
* `getPoints()` may generate the "needed" points, i.e., the points that don't have loaded values and those cannot be used for the quadrature construction
* `getPoints()` and the other `getNeededPoints()` and `getLoadedPoints()` will do work and create a vector, but you only take the size of that vector

The correct method to use here is `grid.getNumLoaded()` which returns an `int` and it is an order 1 operation, i.e., just read a single number.

Reply:
Changed.